README.md

```diff
@@ -2,11 +2,80 @@
 
 Dissertation project for Sheffield Hallam University MSc in Digital & Technology Solutions
 
-#### Overview
+## Overview
 
 The purpose of this application is to compare the read and write performances of 
 a Kafka Streams backed State store against a MongoDB based application.
 
-#### Architecture
+## Architecture
 
-![Architecture Diagram for project](https://github.com/StephenDRoberts/kafka-mongo-comparison/blob/master/assets/KafkaMongoArchitecture.png?raw=true)
\ No newline at end of file
+![Architecture Diagram for project](https://github.com/StephenDRoberts/kafka-mongo-comparison/blob/master/assets/KafkaMongoArchitectureFull.png?raw=true)
+
+## Pre-requisites
+* [Docker](https://www.docker.com/products/docker-desktop)
+* [Docker-compose](https://docs.docker.com/compose/install/)
+
+## Running the applications
+* With docker running on your desktop open a terminal and navigate to the `kafka-mongo-comparison` folder
+* Run `docker-compose up` command to bring up the Kafka, Zookeeper and MongoDB images as well as other external tools for debugging/visual purposes.
+* Run the Setup application. This will only need to be run once. To do so, either:
+    1. navigate to the `setup folder` and run `./gradlew clean bootRun`; or
+    2. run the application through your IDEs run functionality.
+    
+    This will place messages onto the `message-topic` to be read by the other applications.
+* Run your chosen application (either MongoDBApplication or KafkaStreamsApplication). To do so, either:
+    1. navigate to the applications folder and run `./gradlew clean bootRun`; or
+    2. run the application through your IDEs run functionality.
+ 
+    The application will the process all the messages in the `message-topic` kafka topic. 
+    NOTE: once completed, do not stop the application. It will need to stay running for the Poller application to process results.
+* Run the Poller application. This will request statistics from the running application and save them to a CSV file.
+     
+     TODO - dynamic changes between Mongo/KafkaStreams requests (ie ask which application to query, change port number  and excel output workbook name accordingly).
+
+## Setup Module
+
+The purpose of the setup module is to create the experiment's data set.
+This is done via a `@PostConstruct` annotation in the MessageGenerator class that runs a function on setup.
+ 
+
+#### Manually reset offsets
+
+The setup script will create our data set and ideally we only want to run the script once to avoid extra work.
+
+To enable this, we have provided two shell scripts that run as pre-launch configurations for both the MongoDB application, and the Kafka streams version.
```

Comment:
As mentioned on slack `we`. From experience some reviewers can be picky about `I/you/he/she/we/they`

Would just write "to enable this, there are two shell scripts..."